src/backup.py

```diff
@@ -265,6 +265,50 @@ def create_backup(
     return backup_id
 
 
+def restore_backup(
+    container: ops.Container,
+    s3_parameters: S3Parameters,
+    passphrase: str,
+    backup_id: str,
+) -> None:
+    """Restore a backup for Synapse overwriting the current data.
+
+    Args:
+        container: Synapse Container
+        s3_parameters: S3 parameters for the backup.
+        passphrase: Passphrase use to decrypt the backup.
+        backup_id: Name of the object in the backup.
+
+    Raises:
+       BackupError: If there was an error restoring the backup.
+    """
+    _prepare_container(container, s3_parameters, passphrase)
+    container.stop(synapse.SYNAPSE_SERVICE_NAME)
+
+    # Delete the media directory.
+    # Other files to back up will be just overwritten.
+    media_dir = synapse.get_media_store_path(container)
+    if media_dir:
```

Comment:
Is this `if` needed?

Reply:
Interesting point. If that configuration option is not in the homeserver.yaml, I have tested and Synapse sets as default `/media_store` (and Synapse will not start if it does not have permissions there). I have made that explicit in the code and removed the if statement)